### PR TITLE
Update nokogiri to 1.8.1

### PR DESCRIPTION
**Pull request by rcclemente**
⛔️  🚧 I won't be merging this till i can get it to bundle with `classic` 

cc @zendesk/sustaining @zendesk/strongbad 

### Description
Vulnerability Update

### References
* Zd ticket: https://support.zendesk.com/agent/tickets/3062612

### CODE REVIEW

**Comment by craig-day** [comment]
looks like brakeman also needs to be bumped.

**Comment by gfontanilla** [comment]
ZD ticket reference should be https://support.zendesk.com/agent/tickets/3062612 ?

File: lib/abbreviato/abbreviato.rb [hunk 1]
```diff
@@ -22,6 +22,6 @@ def self.truncate(source = '', user_options = {})
     truncated_sax_document = TruncatedSaxDocument.new(DEFAULT_OPTIONS.merge(user_options))
     parser = Nokogiri::HTML::SAX::Parser.new(truncated_sax_document)
     parser.parse(source) { |context| context.replace_entities = false }
-    [truncated_sax_document.truncated_string, truncated_sax_document.truncated]
+    [truncated_sax_document.truncated_string.strip, truncated_sax_document.truncated]
```
Comment (rcclemente) [inline lib/abbreviato/abbreviato.rb:25]: Looks like the parser does not strip trailing spaces so I had to update this to fix the spec test
```ruby
expected: ""
            got: "  "
       (compared using ==)
     # ./spec/support/spec_helpers/abbreviato_macros.rb:6:in `block in test_truncation'
```

File: spec/abbreviato/abbreviato_spec.rb [hunk 2]
```diff
@@ -201,7 +201,7 @@
     let(:cdata_example) { File.read('spec/fixtures/cdata_example.html') }
     it "cdata blocks are preserved" do
       text, truncated = Abbreviato.truncate(cdata_example, max_length: 65535)
-      expect(text.length).to eq 3583
+      expect(text.length).to eq 3581
```
Comment (rcclemente) [inline spec/abbreviato/abbreviato_spec.rb:204]: Leading and trailing spaces got striped

Comment (alduethadyn) [reply spec/abbreviato/abbreviato_spec.rb:204]: @jwsf does seem ok? It jives with what @mesge found I think (the non-stripping with new nokogiri)

Comment (mesge) [reply spec/abbreviato/abbreviato_spec.rb:204]: I suspect this is unrelated to [the thing I found](https://github.com/rubys/nokogumbo/issues/64).

**Review by jwsf — APPROVED** [review]
LGTM 👍

**Review by mariozaizar — APPROVED** [review]

**Pull request merged**